### PR TITLE
docs: move K8s compatability matrix and add more details for compatibility 

### PR DESCRIPTION
**Pull request by david-yu**
Moving the the K8s compatibility matrix to the "Get Started Section".

Adding additional sections

- Vault Secrets Backend Compatibility notes 
- OpenShift Compatibility 
- Detailed change in versioning scheme from 0.33.0+

### CODE REVIEW

File: website/content/docs/k8s/installation/compatibility.mdx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+description: Compatibility Matrix for Consul Kubernetes
+---
+
+# Compatibility Matrix for Consul on Kubernetes 
+
+For every release of Consul on Kubernetes, a Helm chart, `consul-k8s-control-plane` binary and a `consul-k8s` CLI binary is built and distributed through a single version. When deploying via Helm, the recommended best path for upgrading Consul on Kubernetes, is to upgrade using the same `consul-k8s-control-plane` version as the Helm Chart, as the Helm Chart and Control Plane binary are tightly coupled. 
+
+## Supported Consul versions
+
+Starting with Consul Kubernetes 0.33.0, Consul Kubernetes versions all of it components (`conusl-k8s` CLI, `consul-k8s-control-plane`, and Helm chart) with a single version.
```
Comment (ndhanushkodi) [inline website/content/docs/k8s/installation/compatibility.mdx:13]:
```suggestion
Starting with Consul Kubernetes 0.33.0, Consul Kubernetes versions all of it components (`consul-k8s` CLI, `consul-k8s-control-plane`, and Helm chart) with a single version. 
```

Comment (david-yu) [reply website/content/docs/k8s/installation/compatibility.mdx:13]: Thanks!

**Review by thisisnotashwin — APPROVED** [review]
Nicely done!! thanks for doing this David.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/556540.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 331092ead6a2f1b3b7401212a44665ffcbaffb5e onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 331092ead6a2f1b3b7401212a44665ffcbaffb5e onto `release/1.11.x` succeeded!

**Pull request merged**